cmd/kubeadm/app/apis/kubeadm/validation/validation.go

```diff
@@ -594,6 +595,26 @@ func ValidateAPIEndpoint(c *kubeadm.APIEndpoint, fldPath *field.Path) field.Erro
 	return allErrs
 }
 
+// ValidateCertificateKey validates the certificate key is a valid hex encoded AES key
+func ValidateCertificateKey(certificateKey string, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	decodedKey, err := hex.DecodeString(certificateKey)
+	if err != nil {
+		return append(allErrs, field.Invalid(fldPath, certificateKey, fmt.Sprintf("certificate key decoding error: %v", err)))
+	}
+
+	k := len(decodedKey)
+	switch k {
+	case 16, 24, 32:
```

Comment:
our util is flexible to allow various key sizes:
https://github.com/kubernetes/kubernetes/blob/eae63c891882a67b29e08d62cca62686d0635d1a/cmd/kubeadm/app/util/crypto/crypto.go#L38

but we do hardcode the required size to 32 bytes only.
https://github.com/kubernetes/kubernetes/blob/eae63c891882a67b29e08d62cca62686d0635d1a/cmd/kubeadm/app/phases/copycerts/copycerts.go#L81

```
	if len(decodedKey) != kubeadmconstants.CertificateKeySize {
```

```
	allErrs = append(allErrs, field.Invalid(fldPath, certificateKey, fmt.Sprintf("invalid certificate key size %d, the key must be an AES key of size %d", k, kubeadmconstants.CertificateKeySize)))
```

one issue around DecryptBytes is that i don't see size validation impleneted already for it. this means users could have written a valid AES key with smaller size and it would have worked. \..and that's why the comments about validation.go were left i guess.

the main point is that kubeadm by default generates 32 bytes, which is secure.